packages/autocomplete-react/Autocomplete.tsx

```diff
@@ -55,14 +55,21 @@ export function Autocomplete<TItem extends {}>(
         getInputProps={autocomplete.current.getInputProps}
         completion={autocomplete.current.getCompletion()}
         onReset={event => {
-          const { onReset } = autocomplete.current.getResetProps();
+          const { onReset } = autocomplete.current.getFormProps();
           onReset(event);
 
           if (inputRef.current) {
             inputRef.current.focus();
           }
         }}
-        onSubmit={() => {}}
+        onSubmit={event => {
+          const { onSubmit } = autocomplete.current.getFormProps();
+          onSubmit(event);
+
+          if (inputRef.current) {
+            inputRef.current.blur();
```

Comment:
Blurring the input on submit allows to close the keyboard on mobile. Otherwise, it would always stay open until you click on the default "Done" native iOS button.